samples/oneapi/dpc/ccl/sources/decision_forest_reg_hist_distr_ccl.cpp

```diff
@@ -0,0 +1,114 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include <CL/sycl.hpp>
+#include <iomanip>
+#include <iostream>
+
+#ifndef ONEDAL_DATA_PARALLEL
+#define ONEDAL_DATA_PARALLEL
+#endif
+
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/io/csv.hpp"
+#include "oneapi/dal/spmd/ccl/communicator.hpp"
+
+#include "utils.hpp"
+
+namespace dal = oneapi::dal;
+namespace df = dal::decision_forest;
+
+void run(sycl::queue &queue) {
+    const auto train_data_file_name =
+        get_data_path("df_regression_train_data.csv");
+    const auto train_response_file_name =
+        get_data_path("df_regression_train_label.csv");
+    const auto test_data_file_name = get_data_path("df_regression_test_data.csv");
+    const auto test_response_file_name =
+        get_data_path("df_regression_test_label.csv");
+
+    const auto x_train =
+        dal::read<dal::table>(queue, dal::csv::data_source{train_data_file_name});
+    const auto y_train = dal::read<dal::table>(
+        queue, dal::csv::data_source{train_response_file_name});
+    const auto x_test =
+        dal::read<dal::table>(queue, dal::csv::data_source{test_data_file_name});
+    const auto y_test = dal::read<dal::table>(
+        queue, dal::csv::data_source{test_response_file_name});
+
+    auto comm =
+        dal::preview::spmd::make_communicator<dal::preview::spmd::backend::ccl>(
+            queue);
+    auto rank_id = comm.get_rank();
+    auto rank_count = comm.get_rank_count();
+
+    auto x_train_vec = split_table_by_rows<float>(queue, x_train, rank_count);
+    auto y_train_vec = split_table_by_rows<float>(queue, y_train, rank_count);
+    auto x_test_vec = split_table_by_rows<float>(queue, x_test, rank_count);
+    auto y_test_vec = split_table_by_rows<float>(queue, y_test, rank_count);
+
+    const auto df_desc =
+        df::descriptor<float, df::method::hist, df::task::regression>{}
+            .set_tree_count(100)
+            .set_features_per_node(0)
+            .set_min_observations_in_leaf_node(1)
+            .set_error_metric_mode(
+                df::error_metric_mode::out_of_bag_error |
+                df::error_metric_mode::out_of_bag_error_per_observation)
+            .set_variable_importance_mode(df::variable_importance_mode::mdi);
+
+    const auto result_train = dal::preview::train(
+        comm, df_desc, x_train_vec[rank_id], y_train_vec[rank_id]);
+
+    if (rank_id == 0) {
+        std::cout << "Variable importance results:\n"
+                  << result_train.get_var_importance() << std::endl;
+
+        std::cout << "OOB error: " << result_train.get_oob_err() << std::endl;
+        std::cout << "OOB error per observation:\n"
+                  << result_train.get_oob_err_per_observation() << std::endl;
+    }
+
+    const auto result_infer = dal::preview::infer(
+        comm, df_desc, result_train.get_model(), x_test_vec[rank_id]);
+
+    if (comm.get_rank() == 0) {
```

Comment:
```suggestion
     if (rank_id == 0) {
```

Reply:
changed all conditions to if (comm.get_rank() == 0) { to get it in one style